Source/Core/Core/Scripting/HelperClasses/ArgHolder.cpp

```diff
@@ -0,0 +1,237 @@
+#include "Core/Scripting/HelperClasses/ArgHolder.h"
+
+namespace Scripting
+{
+
+ArgHolder* CreateEmptyOptionalArgument()
+{
+  ArgHolder* return_val = new ArgHolder();
+  return_val->contains_value =
+      false;  // We always check if the ArgHolder contains a value before looking at anything else.
```

Comment:
This probably reads nicer if you put the comment on the line before.
```suggestion
  // We always check if the ArgHolder contains a value before looking at anything else.
  return_val->contains_value = false;
```

Reply:
Fixed.